src/servers/http_h2o_server.cc

```diff
@@ -0,0 +1,690 @@
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "src/servers/http_server.h"
+
+#include <errno.h>
+#include <google/protobuf/util/json_util.h>
+#include <h2o.h>
+#include <h2o/cache.h>
+#include <h2o/memcached.h>
+#include <h2o/serverutil.h>
+#include <openssl/ssl.h>
+#include <pthread.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/resource.h>
+#include <sys/signalfd.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include <re2/re2.h>
+#include <algorithm>
+#include <thread>
+#include "src/core/api.pb.h"
+#include "src/core/constants.h"
+#include "src/core/server_status.pb.h"
+#include "src/core/trtserver.h"
+#include "src/servers/common.h"
+
+// The HTTP frontend logging is closely related to the server, thus keep it
+// using the server logging utils
+#include "src/core/logging.h"
+
+#ifdef TRTIS_ENABLE_TRACING
+#include "src/servers/tracer.h"
+#endif  // TRTIS_ENABLE_TRACING
+
+#define USE_HTTPS 0
+#define USE_MEMCACHED 0
+#define DEFAULT_CACHE_LINE_SIZE 128
+
+namespace nvidia { namespace inferenceserver {
+
+// Handle HTTP requests to inference server APIs
+class HTTPAPIServer : public HTTPServer {
+ public:
+  explicit HTTPAPIServer(
+      const std::shared_ptr<TRTSERVER_Server>& server,
+      const std::shared_ptr<nvidia::inferenceserver::TraceManager>&
+          trace_manager,
+      const std::shared_ptr<SharedMemoryBlockManager>& smb_manager,
+      const std::vector<std::string>& endpoints, const int32_t port,
+      const int thread_cnt)
+      : server_(server), trace_manager_(trace_manager),
+        smb_manager_(smb_manager), allocator_(nullptr), port_(port),
+        thread_cnt_(thread_cnt)
+  {
+    TRTSERVER_Error* err = TRTSERVER_ServerId(server_.get(), &server_id_);
+    if (err != nullptr) {
+      server_id_ = "unknown:0";
+      TRTSERVER_ErrorDelete(err);
+    }
+
+    FAIL_IF_ERR(
+        TRTSERVER_ResponseAllocatorNew(
+            &allocator_, ResponseAlloc, ResponseRelease),
+        "creating response allocator");
+  }
+
+  ~HTTPAPIServer()
+  {
+    Stop();
+    LOG_IF_ERR(
+        TRTSERVER_ResponseAllocatorDelete(allocator_),
+        "deleting response allocator");
+  }
+
+ private:
+  TRTSERVER_Error* Start() override;
+  TRTSERVER_Error* Stop() override;
+
+  static TRTSERVER_Error* ResponseAlloc(
+      TRTSERVER_ResponseAllocator* allocator, const char* tensor_name,
+      size_t byte_size, TRTSERVER_Memory_Type preferred_memory_type,
+      int64_t preferred_memory_type_id, void* userp, void** buffer,
+      void** buffer_userp, TRTSERVER_Memory_Type* actual_memory_type,
+      int64_t* actual_memory_type_id);
+  static TRTSERVER_Error* ResponseRelease(
+      TRTSERVER_ResponseAllocator* allocator, void* buffer, void* buffer_userp,
+      size_t byte_size, TRTSERVER_Memory_Type memory_type,
+      int64_t memory_type_id);
+
+  int create_listener(int32_t port);
+#if H2O_USE_LIBUV
```

Comment:
Where is this defined? And is there a reason we need to have 2 choices? If libuv is better we should just use it otherwise we double our testing requirements.

Reply:
Sure, I can remove this. Libuv should be better but I have yet to test this.